spec/compiler/codegen/automatic_cast_spec.cr

```diff
@@ -270,4 +270,15 @@ describe "Code gen: automatic cast" do
       foo(255, 60)
       )).to_i.should eq(255)
   end
+
+  it "casts integer variable to larger type (#9565)" do
+    run(%(
+      def foo(x : Int64)
+        x
+      end
+
+      x = 123
```

Comment:
Should we use an explicitly typed literal here? The test would no longer work if `Int64` became the default int type.

Reply:
All and each of them. Also, this test should be tested with a flag.